crates/primitives/src/proofs.rs

```diff
@@ -73,9 +74,11 @@ pub fn calculate_receipt_root(receipts: &[ReceiptWithBloom]) -> B256 {
     ordered_trie_root_with_encoder(receipts, |r, buf| r.encode_inner(buf, false))
 }
 
-/// Calculate EIP-7685 requests root.
+/// Calculate [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685) requests root.
+///
+/// NOTE: The requests are encoded as `id + request`
```

Comment:
This is a massive footgun and we need to very very clearly articulate it in the spec cc @lightclient